test/SemVer.unittest.js

```diff
@@ -0,0 +1,492 @@
+			"1 < 2",
+			"99 < 100",
+			"1 < 1.2",
+			"1 < 1.2.3",
+			"1.2 < 1.2.3",
+			"1.2.2 < 1.2.3",
+			"1.1.3 < 1.2.0",
+			"1.1.3 < 2.0.0",
+			"1.1.3 < 2",
+			"1.1.3 < 2.0",
+			"1.2.3 < 1.2.3+0",
+			"1.2.3+23 < 1.2.3+123",
+			"1.2+2 < 1.2.3+1",
+			"1.2.3-beta < 1.2.3",
+			"1.2.3 < 1.2.4-beta",
+			"1.2.3 < 1.3.0-beta",
+			"1.2.3 < 2.0.0-beta",
+			"1.2.3-alpha < 1.2.3-beta",
+			"1.2.3-beta < 1.2.3.1",
+			"1.2.3-beta < 1.2.3-beta.0",
+			"1.2.3-beta.0 < 1.2.3-beta.1",
+			"1.2.3-0 < 1.2.3-beta",
+			"1.2.3-beta < 1.2.3-beta+123",
+			"1.2.3-beta+123 < 1.2.3-beta+234",
+			"1.2.3-beta+99 < 1.2.3-beta+111"
+		];
+		for (const c of cases) {
+			it(c, () => {
+				const parts = c.split(" < ");
+				const a = parseVersion(parts[0]);
+				const b = parseVersion(parts[1]);
+				expect(versionLt(a, a)).toBe(false);
+				expect(versionLt(b, b)).toBe(false);
+				expect(versionLt(a, b)).toBe(true);
+				expect(versionLt(b, a)).toBe(false);
+			});
+		}
+	});
+
+	describe("parseRange", () => {
+		const cases = {
+			"=3": ["3", "v3", "3.x", "3.X", "3.x.x", "3.*", "3.*.*", "^3", "^3.x"],
+			"=3.0": ["3.0", "v3.0", "3.0.x", "3.0.X", "3.0.*", "~3.0"],
+			"^3.4": ["^3.4.*"],
+			"3.4 - 6.5": [">=3.4 <=6.5"],
+			"<=3.4": ["<3.4 || =3.4"],
+			">3.4": [">=3.4 !3.4"]
+		};
+		for (const key of Object.keys(cases)) {
+			describe(key, () => {
+				for (const c of cases[key])
+					it(`should be equal ${c}`, () => {
+						expect(parseRange(c)).toEqual(parseRange(key));
+					});
+			});
+		}
+	});
+
+	describe("rangeToString", () => {
+		const cases = {
+			"1": "^1",
+			"1.2": "~1.2",
+			"1.2.3": "=1.2.3",
+			"^1.2.3": "^1.2.3",
+			"~1.2.3": "~1.2.3",
+			"0.0.1": "=0.0.1",
+			"^0.0.1": "=0.0.1",
+			"^0.1.2": "~0.1.2",
+			"~0.0.1": "~0.0.1",
+			"~0.1.2": "~0.1.2",
+			">=1.2.3": ">=1.2.3",
+			"1.2.3-beta.25": "=1.2.3-beta.25",
+			"1.2.3-beta.25+12.34": "=1.2.3-beta.25+12.34",
+			"1.2.3+12.34": "=1.2.3+12.34",
+			">=1.2.3-beta.25": ">=1.2.3-beta.25",
+			">=1.2.3-beta.25+12.34": ">=1.2.3-beta.25+12.34",
+			">=1.2.3+12.34": ">=1.2.3+12.34",
+			"<1.2.3-beta.25": "<1.2.3-beta.25",
+			"<1.2.3-beta.25+12.34": "<1.2.3-beta.25+12.34",
+			"<1.2.3+12.34": "<1.2.3+12.34",
+			"1.2.3 - 3.2.1": ">=1.2.3 (<3.2.1 || =3.2.1)",
+			">3.4": ">=3.4 not(~3.4)",
+			"1 || 2 || 3": "^1 || ^2 || ^3",
+			"1.2.3 - 3.2.1 || >3 <=4 || 1":
+				">=1.2.3 (<3.2.1 || =3.2.1) || >=3 not(^3) (<4 || ^4) || ^1"
+		};
+		for (const key of Object.keys(cases)) {
+			const expected = cases[key];
+			it(`should ${key} stringify to ${expected}`, () => {
+				expect(rangeToString(parseRange(key))).toEqual(expected);
+			});
+		}
+	});
+
+	describe("satisfies", () => {
+		const cases = {
+			// table cases
+			">=1": [
+				"1",
+				"2",
+				"!1-beta",
+				"!2-beta",
+				"1.2",
+				"!1.2-beta",
+				"2.2",
+				"!2.2-beta",
+				"1.beta",
+				"!1.beta-beta",
+				"!2.beta-beta"
+			],
+			">=1-beta": [
+				"1",
+				"2",
+				"1-beta",
+				"!1-gamma",
```

Comment:
In npm semver, `>=1.0.0-beta` includes `1.0.0-gamma`

Reply:
TIL. fix that